api/src/main/java/net/kyori/adventure/text/renderer/AbstractComponentRenderer.java

```diff
@@ -129,6 +133,18 @@
    */
   protected abstract @NotNull Component renderText(final @NotNull TextComponent component, final @NotNull C context);
 
+  /**
+   * Renders a virtual component.
+   *
+   * @param component the component
+   * @param context the context
+   * @return the rendered component
+   * @since 4.13.0
+   */
+  protected @NotNull Component renderVirtual(final @NotNull VirtualComponent component, final @NotNull C context) {
+    return Component.empty();
```

Comment:
should the default here just be a TextComponent with the same style, contents, and children as the virtual component instead?

Reply:
or hm, maybe even pass the component through unchanged just to be safe in multistage rendering processes?

Reply:
yeah - id rather we just do smth ugly like toString or whatever